src/newlines.cpp

```diff
@@ -4883,9 +4887,13 @@ void do_blank_lines(void)
             }
             else if (options::nl_after_func_body() > 0)
             {
-               if (options::nl_after_func_body() != pc->nl_count)
+               // Issue #1734
+               if (!(pc->prev->flags & PCF_IN_TRY_BLOCK))
```

Comment:
BTW, now that #2424 is merged, this can be expressed more explicitly as `if (!pc->prev->flags.test(PCF_IN_TRY_BLOCK))`.